### PR TITLE
Display actually used algorithm name in runtime exception

### PR DESCRIPTION
**Pull request by tamaro-skaljic**
Like #38045 but with requested changes https://github.com/quarkusio/quarkus/pull/38045#discussion_r1444436544.

### CODE REVIEW

**Comment by tamaro-skaljic** [comment]
@gsmet New branch/PR because the last was broken...

File: extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public void run() {
                     PropertiesRealmConfig config = propertiesConfig.file();
                     log.debugf("loadRealm, config=%s", config);
                     SecurityRealm secRealm = realm.getValue();
-                    if (!(secRealm instanceof LegacyPropertiesSecurityRealm)) {
+                    if (!(secRealm instanceof LegacyPropertiesSecurityRealm propsRealm)) {
```
Comment (gastaldi) [inline extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java:62]: I think this is unnecessary, can you please revert? Same as below

Comment (gastaldi) [reply extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java:62]: Ah wait, I see what you done here. The second cast is unused

Comment (gastaldi) [reply extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java:62]: Ignore my comment, I see both are used later in code

**Review by gsmet — APPROVED** [review]
Thanks!

**Pull request merged**